web_src/js/modules/tippy.js

```diff
@@ -25,38 +25,108 @@ export function createTippy(target, opts = {}) {
   return instance;
 }
 
-export function initTooltip(el, props = {}) {
-  const content = el.getAttribute('data-content') || props.content;
+function getTippyTooltipContent(target) {
+  // prefer to always use the "[data-tooltip-content]" attribute
+  // for backward compatibility, we also support the ".tooltip[data-content]" attribute
+  let content = target.getAttribute('data-tooltip-content');
+  if (!content && target.classList.contains('tooltip')) {
+    content = target.getAttribute('data-content');
+  }
+  return content;
+}
+
+/**
+ * Attach a tippy tooltip to the given target element.
+ * If the target element already has a tippy tooltip attached, the tooltip will be updated with the new content.
+ * If the target element has no content, then no tooltip will be attached, and it returns null.
+ * @param target {HTMLElement}
+ * @param content {null|string}
+ * @returns {null|tippy}
+ */
+function attachTippyTooltip(target, content = null) {
+  content = content ?? getTippyTooltipContent(target);
   if (!content) return null;
-  if (!el.hasAttribute('aria-label')) el.setAttribute('aria-label', content);
-  return createTippy(el, {
+
+  const props = {
     content,
     delay: 100,
     role: 'tooltip',
-    ...(el.getAttribute('data-tooltip-interactive') === 'true' ? {interactive: true} : {}),
-    ...props,
-  });
-}
+    placement: target.getAttribute('data-tooltip-placement') || 'top-start',
+    ...(target.getAttribute('data-tooltip-interactive') === 'true' ? {interactive: true} : {}),
```

Comment:
I'd say we move these `getAttribute` checks into `createTippy` and rename the attributes to `data-tippy-placement` and `data-tippy-interactive`, renaming the previous `data-placement` as well.

Reply:
I have thought about it, however, I didn't choose this approach. The reason is:

1. `tooltip` doesn't mean `tippy`. The tooltip is only "auto-show" tippy, so the concept shouldn't be mixed together.
2. The `createTippy` calls are all by JS code, no need to do more assume from `data-` attributes.


A whole picture in my mind is:

1. If we want to show (auto) tooltip, write some `data-tooltip-xxx` attributes.
2. If we want to show some tippy, call `createTippy` manually, and prepare necessary props by JS code.


Think about a more complex case: if we have a button, it has tooltip, and it need to show a custom tippy popup?

Reply:
Hmm, at least convert `data-placement` in `templates/repo/sub_menu.tmpl` to `data-tooltip-placement` and remove the check for it inside `createTippy`.

Reply:
I believe I have  done that in the first commit of this PR 

https://github.com/go-gitea/gitea/pull/23574/commits/d06b47cccdd1a27311a6f3342f80a139e0355299#diff-5e8ef0864f77e933281dd9cd0709637d865015e3fa7944aefcf2cb6e1480f8d4

There is  no `data-placement` in templates after this PR.

Reply:
Ah yes indeed, I missed that.